src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelDepGraphFunction.java

```diff
@@ -128,18 +128,18 @@ public SkyValue compute(SkyKey skyKey, Environment env)
         calculateUniqueNameForUsedExtensionId(extensionUsagesById);
 
     if (lockfileMode.equals(LockfileMode.UPDATE)) {
-      // This will keep all module extension evaluation results, some of which may be stale due to
-      // changed usages. They will be removed in BazelLockFileModule.
-      BazelLockFileValue updateLockfile =
-          lockfile.toBuilder()
-              .setLockFileVersion(BazelLockFileValue.LOCK_FILE_VERSION)
+      BazelLockFileValue resolutionOnlyLockfile =
+          BazelLockFileValue.builder()
               .setModuleFileHash(root.getModuleFileHash())
               .setFlags(flags)
               .setLocalOverrideHashes(localOverrideHashes)
               .setModuleDepGraph(depGraph)
               .build();
+      Preconditions.checkNotNull(lockfile);
```

Comment:
Why this check is needed?
If the lockfile mode is update, it should never be null.

Reply:
Yes, that's what I wanted to document with this precondition. Ideally I could remove the `= null` from line 72, but that's not something the compiler can verify is unused. I can document this in a different way or just remove the line.

Reply:
I just think it may indicate that "it may be null" and this will never happen. So maybe another way? or removing it are both fine.

Reply:
I removed it!